// @antv/s2-vue/esm/components/tooltip/components/operator/menu.vue2.js
import _sfc_main from "./menu.vue.js";
import { resolveComponent, createElementBlock, openBlock, Fragment, createBlock, createCommentVNode, withCtx, renderList, createVNode } from "vue";
import _export_sfc from "../../../../_virtual/_plugin-vue_export-helper.js";
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_TooltipOperatorTitle = resolveComponent("TooltipOperatorTitle");
  const _component_TooltipOperatorMenu = resolveComponent("TooltipOperatorMenu");
  const _component_MenuItem = resolveComponent("MenuItem");
  const _component_SubMenu = resolveComponent("SubMenu");
  return openBlock(), createElementBlock(Fragment, null, [
    !_ctx.isEmpty(_ctx.menu.children) ? (openBlock(), createBlock(_component_SubMenu, {
      key: _ctx.menu.key,
      popupClassName: `${_ctx.TOOLTIP_PREFIX_CLS}-operator-submenu-popup`,
      onTitleClick: _ctx.onMenuTitleClick
    }, {
      title: withCtx(() => [
        createVNode(_component_TooltipOperatorTitle, {
          menu: _ctx.menu,
          onClick: _ctx.onMenuTitleClick
        }, null, 8, ["menu", "onClick"])
      ]),
      default: withCtx(() => [
        (openBlock(true), createElementBlock(Fragment, null, renderList(_ctx.menu.children, (subMenu) => {
          var _a;
          return openBlock(), createElementBlock(Fragment, {
            key: subMenu.key
          }, [
            ((_a = subMenu == null ? void 0 : subMenu.children) == null ? void 0 : _a.length) ? (openBlock(), createBlock(_component_TooltipOperatorMenu, {
              key: 0,
              menu: subMenu,
              cell: _ctx.cell
            }, null, 8, ["menu", "cell"])) : (openBlock(), createBlock(_component_MenuItem, {
              title: subMenu.label,
              key: subMenu.key
            }, {
              default: withCtx(() => [
                createVNode(_component_TooltipOperatorTitle, {
                  menu: subMenu,
                  onClick: _ctx.onMenuTitleClick
                }, null, 8, ["menu", "onClick"])
              ]),
              _: 2
            }, 1032, ["title"]))
          ], 64);
        }), 128))
      ]),
      _: 1
    }, 8, ["popupClassName", "onTitleClick"])) : createCommentVNode("", true),
    _ctx.isEmpty(_ctx.menu.children) ? (openBlock(), createBlock(_component_MenuItem, {
      title: _ctx.menu.label,
      key: _ctx.menu.key
    }, {
      default: withCtx(() => [
        createVNode(_component_TooltipOperatorTitle, {
          menu: _ctx.menu,
          onClick: _ctx.onMenuTitleClick
        }, null, 8, ["menu", "onClick"])
      ]),
      _: 1
    }, 8, ["title"])) : createCommentVNode("", true)
  ], 64);
}
const TooltipOperatorMenu = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  TooltipOperatorMenu as default
};
