// @antv/s2-vue/lib/components/sheets/grid-analysis-sheet/index.vue2.js
"use strict";
const s2 = require("@antv/s2");
const vue = require("vue");
const useExpose = require("../../../hooks/useExpose.js");
const initPropAndEmits = require("../../../utils/initPropAndEmits.js");
const baseSheet = require("../base-sheet.vue.js");
const customCell = require("./custom-cell.js");
const theme = require("./theme.js");
const _sfc_main = vue.defineComponent({
  name: "GridAnalysisSheet",
  props: initPropAndEmits.initBaseSheetProps(),
  emits: [],
  setup(props, ctx) {
    const s2Ref = useExpose.useExpose(ctx.expose);
    const { options: defaultOptions, themeCfg } = vue.toRefs(props);
    const s2Options = vue.computed(() => {
      const options = {
        dataCell: (viewMeta, spreadsheet) => new customCell.GridAnalysisSheetDataCell(viewMeta, spreadsheet),
        showDefaultHeaderActionIcon: false,
        style: {
          colCell: {
            hideValue: true
          }
        }
      };
      return s2.customMerge(defaultOptions == null ? void 0 : defaultOptions.value, options);
    });
    const s2ThemeCfg = vue.computed(
      () => s2.customMerge(themeCfg == null ? void 0 : themeCfg.value, { theme: theme.GridAnalysisTheme })
    );
    return {
      s2Ref,
      s2Options,
      s2ThemeCfg
    };
  },
  components: {
    BaseSheet: baseSheet
  }
});
module.exports = _sfc_main;
