// @antv/s2-vue/lib/components/sheets/grid-analysis-sheet/theme.js
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const s2 = require("@antv/s2");
const FONT_SIZE = 12;
const FONT_SIZE_MINOR = 11;
const GridAnalysisTheme = {
  // 表头
  dataCell: {
    text: {
      fontSize: FONT_SIZE,
      fontFamily: s2.FONT_FAMILY,
      fill: "rgba(0, 0, 0, 0.85)",
      fontWeight: "normal",
      textBaseline: "middle",
      textAlign: "left"
    },
    // 次级文本，如副指标
    minorText: {
      fontSize: FONT_SIZE_MINOR,
      fontFamily: s2.FONT_FAMILY,
      fill: "rgba(0, 0, 0, 0.45)",
      fontWeight: "normal",
      textBaseline: "middle",
      textAlign: "left"
    },
    // 衍生指标
    derivedMeasureText: {
      mainUp: "#F46649",
      mainDown: "#2AA491",
      minorUp: "#f9ae9e",
      minorDown: "#a9dad2"
    },
    bolderText: {
      fontSize: FONT_SIZE,
      fontFamily: s2.FONT_FAMILY,
      fill: "rgba(0, 0, 0, 1)",
      fontWeight: s2.isWindows() ? "bold" : 520,
      textBaseline: "middle",
      textAlign: "left"
    }
  }
};
exports.FONT_SIZE = FONT_SIZE;
exports.FONT_SIZE_MINOR = FONT_SIZE_MINOR;
exports.GridAnalysisTheme = GridAnalysisTheme;
