// @antv/s2-vue/esm/components/pagination/index.vue.js
import _sfc_main from "./index.vue2.js";
import { resolveComponent, createElementBlock, openBlock, normalizeClass, createVNode, createElementVNode, mergeProps, toDisplayString } from "vue";
import _export_sfc from "../../_virtual/_plugin-vue_export-helper.js";
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_AntDPagination = resolveComponent("AntDPagination");
  return openBlock(), createElementBlock("div", {
    class: normalizeClass(_ctx.PRE_CLASS)
  }, [
    createVNode(_component_AntDPagination, mergeProps({
      size: "small",
      "default-current": 1,
      showSizeChanger: true,
      showQuickJumper: _ctx.showQuickJumper
    }, _ctx.customOptions, {
      current: _ctx.current,
      total: _ctx.total,
      pageSize: _ctx.pageSize,
      onShowSizeChange: _cache[0] || (_cache[0] = (_, size) => _ctx.$emit("showSizeChange", size)),
      onChange: _cache[1] || (_cache[1] = (current) => _ctx.$emit("change", current))
    }), null, 16, ["showQuickJumper", "current", "total", "pageSize"]),
    createElementVNode("span", {
      class: normalizeClass(`${_ctx.PRE_CLASS}-count`)
    }, toDisplayString(_ctx.i18n("共计")) + " " + toDisplayString(_ctx.total || " - ") + " " + toDisplayString(_ctx.i18n("条")), 3)
  ], 2);
}
const S2Pagination = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  S2Pagination as default
};
