// @antv/s2-vue/lib/components/sheets/editable-sheet/drag-copy/DragCopyMask.vue2.js
"use strict";
const DragCopyMask_vue_vue_type_script_lang = require("./DragCopyMask.vue.js");
const vue = require("vue");
const _pluginVue_exportHelper = require("../../../../_virtual/_plugin-vue_export-helper.js");
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return vue.openBlock(), vue.createElementBlock("div", {
    class: vue.normalizeClass(`${_ctx.S2_PREFIX_CLS}-drag-copy-mask`),
    style: vue.normalizeStyle({
      right: _ctx.maskPosition.right > 0 ? `${_ctx.maskPosition.right}px` : "0",
      bottom: _ctx.maskPosition.bottom > 0 ? `${_ctx.maskPosition.bottom}px` : "0",
      left: _ctx.maskPosition.right <= 0 ? `${_ctx.maskPosition.right}px` : "0",
      top: _ctx.maskPosition.bottom <= 0 ? `${_ctx.maskPosition.bottom}px` : "0",
      width: `${Math.abs(_ctx.maskPosition.right)}px`,
      height: `${Math.abs(_ctx.maskPosition.bottom)}px`
    })
  }, null, 6);
}
const DragCopyMask = /* @__PURE__ */ _pluginVue_exportHelper(DragCopyMask_vue_vue_type_script_lang, [["render", _sfc_render]]);
module.exports = DragCopyMask;
