// @antv/s2-vue/lib/components/sheets/strategy-sheet/custom-tooltip/col-cell-tooltip.vue2.js
"use strict";
const s2 = require("@antv/s2");
const lodash = require("lodash");
const vue = require("vue");
const _sfc_main = vue.defineComponent({
  name: "StrategySheetColCellTooltip",
  props: {
    cell: {
      type: Object,
      required: true
    },
    label: {
      type: [Object, Function],
      default: void 0
    }
  },
  setup(props) {
    const meta = vue.computed(() => props.cell.getMeta());
    const shouldHide = vue.computed(() => {
      return meta.value.isLeaf && meta.value.level !== 0;
    });
    const cellName = vue.computed(() => {
      const m = meta.value;
      const name = m.spreadsheet.dataSet.getFieldName(m.field);
      const customLabel = lodash.isFunction(props.label) ? props.label(props.cell, name) : props.label;
      return customLabel != null ? customLabel : name;
    });
    const cellValue = vue.computed(() => meta.value.value);
    return {
      tooltipCls: s2.getStrategySheetTooltipClsName,
      shouldHide,
      cellName,
      cellValue
    };
  }
});
module.exports = _sfc_main;
