// @antv/s2-vue/lib/components/sheets/strategy-sheet/custom-tooltip/data-cell-tooltip.vue2.js
"use strict";
const s2 = require("@antv/s2");
const lodash = require("lodash");
const vue = require("vue");
const _sfc_main = vue.defineComponent({
  name: "StrategySheetDataCellTooltip",
  props: {
    cell: {
      type: Object,
      required: true
    },
    label: {
      type: [Object, Function],
      default: void 0
    },
    showOriginalValue: {
      type: Boolean,
      default: false
    },
    renderDerivedValue: {
      type: Function,
      default: void 0
    }
  },
  setup(props) {
    const meta = vue.computed(() => props.cell.getMeta());
    const spreadsheet = vue.computed(() => meta.value.spreadsheet);
    const metaFieldValue = vue.computed(
      () => {
        var _a;
        return (_a = meta.value) == null ? void 0 : _a.fieldValue;
      }
    );
    const rowDescription = vue.computed(
      () => spreadsheet.value.dataSet.getCustomFieldDescription(props.cell)
    );
    const rowName = vue.computed(() => {
      const defaultRowName = spreadsheet.value.dataSet.getCustomRowFieldName(
        props.cell
      );
      const customLabel = lodash.isFunction(props.label) ? props.label(props.cell, defaultRowName) : props.label;
      return customLabel != null ? customLabel : defaultRowName;
    });
    const colLeafNode = vue.computed(
      () => spreadsheet.value.facet.getColLeafNodeByIndex(meta.value.colIndex)
    );
    const derivedLabels = vue.computed(() => {
      var _a;
      try {
        const [, ...labels] = JSON.parse((_a = colLeafNode.value) == null ? void 0 : _a.value);
        return labels;
      } catch (e) {
        return [];
      }
    });
    const valuesCfg = vue.computed(
      () => {
        var _a, _b;
        return (_b = (_a = spreadsheet.value.options.style) == null ? void 0 : _a.dataCell) == null ? void 0 : _b.valuesCfg;
      }
    );
    const values = vue.computed(() => {
      var _a;
      const firstValues = lodash.first((_a = metaFieldValue.value) == null ? void 0 : _a.values) || [
        metaFieldValue.value
      ];
      return {
        value: firstValues[0],
        derivedValues: firstValues.slice(1)
      };
    });
    const originalValues = vue.computed(() => {
      var _a;
      const originalValuesField = lodash.get(
        metaFieldValue.value,
        (_a = valuesCfg.value) == null ? void 0 : _a.originalValueField
      );
      const firstOriginal = lodash.first(originalValuesField) || [values.value.value];
      return {
        originalValue: firstOriginal[0],
        derivedOriginalValues: firstOriginal.slice(1)
      };
    });
    const emptyPlaceholder = vue.computed(
      () => s2.getEmptyPlaceholder(meta.value, spreadsheet.value.options.placeholder)
    );
    const shouldShowOriginalValue = vue.computed(
      () => {
        var _a;
        return ((_a = valuesCfg.value) == null ? void 0 : _a.showOriginalValue) || props.showOriginalValue;
      }
    );
    const getDerivedValueInfo = (derivedValue, index) => {
      const isUnchanged = s2.isUnchangedValue(
        derivedValue,
        values.value.value
      );
      const isUp = !isUnchanged && s2.isUpDataValue(derivedValue);
      const isDown = !isUnchanged && !isUp;
      const originalDerivedValue = originalValues.value.derivedOriginalValues[index];
      return { isUnchanged, isUp, isDown, originalDerivedValue };
    };
    return {
      tooltipCls: s2.getStrategySheetTooltipClsName,
      i18n: s2.i18n,
      isEmpty: lodash.isEmpty,
      isNil: lodash.isNil,
      rowName,
      rowDescription,
      values,
      originalValues,
      derivedLabels,
      emptyPlaceholder,
      shouldShowOriginalValue,
      getDerivedValueInfo
    };
  }
});
module.exports = _sfc_main;
