// @antv/s2-vue/lib/components/sheets/strategy-sheet/custom-tooltip/row-cell-tooltip.vue2.js
"use strict";
const s2 = require("@antv/s2");
const lodash = require("lodash");
const vue = require("vue");
const _sfc_main = vue.defineComponent({
  name: "StrategySheetRowCellTooltip",
  props: {
    cell: {
      type: Object,
      required: true
    },
    label: {
      type: [Object, Function],
      default: void 0
    }
  },
  setup(props) {
    const meta = vue.computed(() => props.cell.getMeta());
    const rowName = vue.computed(() => {
      const { value } = meta.value;
      const customLabel = lodash.isFunction(props.label) ? props.label(props.cell, value) : props.label;
      return customLabel != null ? customLabel : value;
    });
    const description = vue.computed(() => {
      const { field, spreadsheet, extra } = meta.value;
      return spreadsheet.dataSet.getFieldDescription(field) || (extra == null ? void 0 : extra["description"]);
    });
    return {
      tooltipCls: s2.getStrategySheetTooltipClsName,
      i18n: s2.i18n,
      rowName,
      description
    };
  }
});
module.exports = _sfc_main;
