// @antv/s2-vue/esm/components/sheets/base-sheet.vue.js
import _sfc_main from "./base-sheet.vue2.js";
import { resolveComponent, createBlock, openBlock, withCtx, createElementVNode, normalizeClass, createCommentVNode, renderSlot } from "vue";
/* empty css                */
import _export_sfc from "../../_virtual/_plugin-vue_export-helper.js";
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_S2Pagination = resolveComponent("S2Pagination");
  const _component_Spin = resolveComponent("Spin");
  return openBlock(), createBlock(_component_Spin, {
    wrapperClassName: _ctx.S2_PREFIX_CLS + "-spin",
    spinning: _ctx.loading
  }, {
    default: withCtx(() => [
      createElementVNode("div", {
        ref: "wrapperRef",
        class: normalizeClass(_ctx.S2_PREFIX_CLS + "-wrapper")
      }, [
        createElementVNode("div", {
          ref: "containerRef",
          class: normalizeClass(_ctx.S2_PREFIX_CLS + "-container")
        }, null, 2),
        _ctx.pagination.visible.value ? (openBlock(), createBlock(_component_S2Pagination, {
          key: 0,
          current: _ctx.pagination.current.value,
          pageSize: _ctx.pagination.pageSize.value,
          total: _ctx.pagination.total.value,
          customOptions: _ctx.pagination.customOptions,
          onChange: _ctx.handlePageChange,
          onShowSizeChange: _ctx.handlePageSizeChange
        }, null, 8, ["current", "pageSize", "total", "customOptions", "onChange", "onShowSizeChange"])) : createCommentVNode("", true),
        renderSlot(_ctx.$slots, "editCell")
      ], 2)
    ]),
    _: 3
  }, 8, ["wrapperClassName", "spinning"]);
}
const BaseSheet = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  BaseSheet as default
};
