// @antv/s2-vue/esm/components/sheets/base-sheet.vue2.js
import { S2_PREFIX_CLS } from "@antv/s2";
import { Spin } from "ant-design-vue";
import { defineComponent, computed } from "vue";
import { isBoolean } from "lodash";
import S2Pagination from "../pagination/index.vue.js";
import { provideSpreadSheet } from "../../context/SpreadSheetContext.js";
import { useSpreadSheet } from "../../hooks/useSpreadSheet.js";
import { initBaseSheetEmits, initBaseSheetProps } from "../../utils/initPropAndEmits.js";
const _sfc_main = defineComponent({
  name: "BaseSheet",
  props: initBaseSheetProps(),
  emits: initBaseSheetEmits(),
  setup(props, ctx) {
    const { wrapperRef, containerRef, s2Ref, loading, pagination } = useSpreadSheet(props, ctx.emit);
    const s2RefComputed = computed(() => {
      var _a;
      return (_a = s2Ref.value) != null ? _a : null;
    });
    provideSpreadSheet(s2RefComputed);
    ctx.expose({
      get instance() {
        return s2Ref.value;
      }
    });
    const handlePageChange = (nextCurrent) => {
      var _a, _b;
      if (props.showPagination && !isBoolean(props.showPagination)) {
        (_b = (_a = props.showPagination).onChange) == null ? void 0 : _b.call(_a, nextCurrent);
      }
      pagination.change(nextCurrent);
    };
    const handlePageSizeChange = (nextSize) => {
      var _a, _b;
      if (props.showPagination && !isBoolean(props.showPagination)) {
        (_b = (_a = props.showPagination).onShowSizeChange) == null ? void 0 : _b.call(_a, nextSize);
      }
      pagination.showSizeChange(nextSize);
    };
    return {
      S2_PREFIX_CLS,
      wrapperRef,
      containerRef,
      s2Ref,
      loading,
      pagination,
      handlePageChange,
      handlePageSizeChange
    };
  },
  components: {
    Spin,
    S2Pagination
  }
});
export {
  _sfc_main as default
};
