// @antv/s2-vue/esm/components/sheets/chart-sheet.vue2.js
import { customMerge } from "@antv/s2";
import { defineComponent, toRefs, computed } from "vue";
import { useExpose } from "../../hooks/useExpose.js";
import { initBaseSheetProps } from "../../utils/initPropAndEmits.js";
import BaseSheet from "./base-sheet.vue.js";
const _sfc_main = defineComponent({
  name: "ChartSheet",
  props: initBaseSheetProps(),
  emits: [],
  setup(props, ctx) {
    const s2Ref = useExpose(ctx.expose);
    const { options: chartOptions, themeCfg: customThemeCfg } = toRefs(props);
    const s2Options = computed(() => {
      const defaultOptions = {
        style: {
          rowCell: {
            width: 100
          },
          dataCell: {
            width: 400,
            height: 400
          }
        },
        tooltip: {
          enable: true
        }
      };
      const options = {
        showDefaultHeaderActionIcon: false,
        interaction: {
          hoverFocus: false,
          brushSelection: {
            dataCell: false
          }
        }
      };
      return customMerge(
        defaultOptions,
        chartOptions.value,
        options
      );
    });
    const themeCfg = computed(() => {
      const defaultTheme = {
        dataCell: {
          cell: {
            interactionState: {
              hoverFocus: {
                borderOpacity: 0
              },
              selected: {
                borderOpacity: 0
              }
            }
          }
        }
      };
      return customMerge(defaultTheme, customThemeCfg == null ? void 0 : customThemeCfg.value);
    });
    return {
      s2Ref,
      s2Options,
      themeCfg
    };
  },
  components: {
    BaseSheet
  }
});
export {
  _sfc_main as default
};
