// @antv/s2-vue/esm/components/sheets/editable-sheet.vue2.js
var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
import { defineComponent, toRefs, ref, computed, reactive, watch } from "vue";
import { Input } from "ant-design-vue";
import { pick, isNil } from "lodash";
import { useExpose } from "../../hooks/useExpose.js";
import { initBaseSheetProps } from "../../utils/initPropAndEmits.js";
import BaseSheet from "./base-sheet.vue.js";
import "./editable-sheet/drag-copy/DragCopyMask.vue.js";
import DragCopyPoint from "./editable-sheet/drag-copy/DragCopyPoint.vue.js";
function buildEditProps(option) {
  return __spreadValues({}, option);
}
const _sfc_main = defineComponent({
  name: "EditableSheet",
  props: initBaseSheetProps(),
  emits: [],
  setup(props, ctx) {
    var _a;
    const s2Ref = useExpose(ctx.expose);
    const { options: originOptions } = toRefs(props);
    const inputRef = ref(null);
    const targetCell = ref(null);
    const inputValue = ref("");
    const options = computed(
      () => buildEditProps(originOptions.value)
    );
    const inputStyle = reactive({
      width: 0,
      height: 0,
      left: 0,
      top: 0,
      display: "none",
      zIndex: 1e3,
      position: "absolute"
    });
    function setInputStyle() {
      var _a2;
      const spreadsheet = (_a2 = s2Ref.value) == null ? void 0 : _a2.instance;
      const cell = targetCell.value;
      if (!spreadsheet || !cell) {
        inputStyle.display = "none";
        return;
      }
      const scroll = spreadsheet.facet.getScrollOffset();
      let cellPosition = pick(cell.getMeta(), ["x", "y", "width", "height"]);
      if (isNil(cellPosition.x) || isNil(cellPosition.y)) {
        cellPosition = {
          x: 0,
          y: 0,
          width: 0,
          height: 0
        };
      }
      const sampleColNode = spreadsheet.facet.getColNodes()[0];
      const sampleColNodeHeight = (sampleColNode == null ? void 0 : sampleColNode.height) || 0;
      cellPosition.x -= scroll.scrollX || 0;
      cellPosition.y -= (scroll.scrollY || 0) - sampleColNodeHeight;
      const {
        x: cellLeft,
        y: cellTop,
        width: cellWidth,
        height: cellHeight
      } = cellPosition;
      const inSight = cellTop >= 0 && cellTop <= spreadsheet.facet.getCanvasSize().height && cellLeft >= 0 && cellLeft <= spreadsheet.facet.getCanvasSize().width;
      inputStyle.width = cellWidth ? `${cellWidth}px` : "0px";
      inputStyle.height = cellHeight ? `${cellHeight}px` : "0px";
      inputStyle.left = cellLeft ? `${cellLeft}px` : "0px";
      inputStyle.top = cellTop ? `${cellTop}px` : "0px";
      inputStyle.display = targetCell.value && inSight ? "block" : "none";
    }
    watch([targetCell, (_a = s2Ref.value) == null ? void 0 : _a.instance.facet.getScrollOffset()], () => {
      setInputStyle();
    });
    const onDataCellDbClick = (cell) => {
      targetCell.value = cell.target;
      inputValue.value = cell.target.getActualText();
      setTimeout(() => {
        var _a2;
        (_a2 = inputRef.value) == null ? void 0 : _a2.focus();
      }, 100);
    };
    function onSave() {
      var _a2, _b;
      const target = inputRef.value;
      const cell = targetCell.value;
      const spreadsheet = (_a2 = s2Ref.value) == null ? void 0 : _a2.instance;
      if (!spreadsheet || !cell || !target) {
        return;
      }
      const { rowIndex, valueField, id } = cell.getMeta();
      const inputVal = target.value;
      const displayData = spreadsheet.dataSet.getDisplayDataSet();
      displayData[rowIndex][valueField] = inputVal;
      (_b = spreadsheet.dataSet.displayFormattedValueMap) == null ? void 0 : _b.set(id, inputVal);
      spreadsheet.render();
      targetCell.value = null;
    }
    return {
      setInputStyle,
      s2Ref,
      options,
      inputStyle,
      onDataCellDbClick,
      inputRef,
      onSave,
      inputValue,
      Input
    };
  },
  components: {
    BaseSheet,
    DragCopyPoint
  }
});
export {
  _sfc_main as default
};
