// @antv/s2-vue/esm/components/sheets/editable-sheet.vue.js
import _sfc_main from "./editable-sheet.vue2.js";
import { resolveComponent, createBlock, openBlock, mergeProps, withCtx, createVNode, normalizeStyle } from "vue";
import _export_sfc from "../../_virtual/_plugin-vue_export-helper.js";
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_Input = resolveComponent("Input");
  const _component_DragCopyPoint = resolveComponent("DragCopyPoint");
  const _component_BaseSheet = resolveComponent("BaseSheet");
  return openBlock(), createBlock(_component_BaseSheet, mergeProps({
    onDataCellDoubleClick: _ctx.onDataCellDbClick,
    onScroll: _ctx.setInputStyle,
    onDataCellClick: _ctx.onSave,
    ref: "s2Ref"
  }, _ctx.$props, { options: _ctx.options }), {
    editCell: withCtx(() => [
      createVNode(_component_Input, {
        onBlur: _ctx.onSave,
        value: _ctx.inputValue,
        ref: "inputRef",
        style: normalizeStyle(_ctx.inputStyle)
      }, null, 8, ["onBlur", "value", "style"]),
      createVNode(_component_DragCopyPoint)
    ]),
    _: 1
  }, 16, ["onDataCellDoubleClick", "onScroll", "onDataCellClick", "options"]);
}
const EditableSheet = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  EditableSheet as default
};
