// @antv/s2-vue/esm/components/sheets/pivot-sheet.vue2.js
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
import { buildDrillDownOptions, handleDrillDown } from "@antv/s2";
import { isEmpty } from "lodash";
import { defineComponent, toRefs, ref, computed, createVNode } from "vue";
import { useExpose } from "../../hooks/useExpose.js";
import { initBaseSheetProps } from "../../utils/initPropAndEmits.js";
import DrillDown from "../drill-down/index.vue.js";
import BaseSheet from "./base-sheet.vue.js";
const _sfc_main = defineComponent({
  name: "PivotSheet",
  props: initBaseSheetProps(),
  emits: [],
  setup(props, ctx) {
    const s2Ref = useExpose(ctx.expose);
    const { options: pivotOptions } = toRefs(props);
    const { dataCfg, partDrillDown } = toRefs(props);
    const drillFields = ref([]);
    const setDrillFields = (fields) => {
      var _a, _b, _c, _d, _e, _f;
      const instance = (_a = s2Ref == null ? void 0 : s2Ref.value) == null ? void 0 : _a.instance;
      drillFields.value = fields;
      instance == null ? void 0 : instance.hideTooltip();
      if (isEmpty(fields)) {
        instance == null ? void 0 : instance.clearDrillDownData((_b = instance == null ? void 0 : instance.store.get("drillDownNode")) == null ? void 0 : _b.id);
      } else {
        handleDrillDown({
          rows: (_d = (_c = dataCfg.value) == null ? void 0 : _c.fields.rows) != null ? _d : [],
          drillFields: fields,
          fetchData: (_e = partDrillDown.value) == null ? void 0 : _e["fetchData"],
          drillItemsNum: (_f = partDrillDown.value) == null ? void 0 : _f["drillItemsNum"],
          spreadsheet: instance
        });
      }
    };
    const onDrillDownIconClick = (params) => {
      var _a, _b;
      const { event, disabledFields } = params;
      if (event) {
        const instance = (_a = s2Ref == null ? void 0 : s2Ref.value) == null ? void 0 : _a.instance;
        const drillDownNode = createVNode(DrillDown, __spreadProps(__spreadValues({}, (_b = partDrillDown.value) == null ? void 0 : _b["drillConfig"]), {
          setDrillFields,
          drillFields: drillFields.value,
          disabledFields
        }));
        instance == null ? void 0 : instance.showTooltip({
          position: {
            x: event.clientX,
            y: event.clientY
          },
          event,
          content: drillDownNode
        });
      }
    };
    const options = computed(
      () => buildDrillDownOptions(
        pivotOptions.value,
        partDrillDown.value,
        (params) => onDrillDownIconClick(params)
      )
    );
    return {
      s2Ref,
      options
    };
  },
  components: {
    BaseSheet
  }
});
export {
  _sfc_main as default
};
