// @antv/s2-vue/esm/utils/initPropAndEmits.js
import { i18n } from "@antv/s2";
const initBaseSheetProps = () => {
  return {
    sheetType: String,
    dataCfg: Object,
    themeCfg: Object,
    showPagination: {
      type: [Object, Boolean],
      default: false
    },
    loading: Boolean,
    partDrillDown: Object,
    header: Object,
    options: {
      type: Object,
      default: {}
    },
    adaptive: {
      type: [Object, Boolean],
      default: false
    },
    spreadsheet: Function,
    onMounted: Function,
    onUpdate: Function,
    onUpdateAfterRender: Function
  };
};
const initDrillDownProps = () => {
  return {
    className: String,
    title: {
      type: String,
      default: i18n("选择下钻维度")
    },
    searchText: {
      type: String,
      default: i18n("搜索字段")
    },
    clearText: {
      type: String,
      default: i18n("恢复默认")
    },
    extra: Node,
    dataSet: {
      type: Array,
      default: []
    },
    drillFields: Array,
    disabledFields: Array,
    getDrillFields: Function,
    setDrillFields: Function
  };
};
const initDrillDownEmits = () => {
  const keys = ["getDrillFields", "setDrillFields"];
  return keys;
};
const initBaseSheetEmits = () => {
  const keys = [
    "spreadsheet",
    // ============== Row Cell ====================
    "rowCellHover",
    "rowCellClick",
    "rowCellDoubleClick",
    "rowCellContextMenu",
    "rowCellMouseDown",
    "rowCellMouseUp",
    "rowCellMouseMove",
    "rowCellCollapsed",
    "rowCellAllCollapsed",
    "rowCellScroll",
    // ============== Col Cell ====================
    "colCellHover",
    "colCellClick",
    "colCellDoubleClick",
    "colCellContextMenu",
    "colCellMouseDown",
    "colCellMouseUp",
    "colCellMouseMove",
    "colCellExpanded",
    "colCellHidden",
    // ============== Data Cell ====================
    "dataCellHover",
    "dataCellClick",
    "dataCellDoubleClick",
    "dataCellContextMenu",
    "dataCellMouseDown",
    "dataCellMouseUp",
    "dataCellMouseMove",
    "dataCellBrushSelection",
    "dataCellSelectMove",
    // ============== Corner Cell ====================
    "cornerCellHover",
    "cornerCellClick",
    "cornerCellDoubleClick",
    "cornerCellContextMenu",
    "cornerCellMouseDown",
    "cornerCellMouseUp",
    "cornerCellMouseMove",
    // ============== Merged Cells ====================
    "mergedCellsHover",
    "mergedCellsClick",
    "mergedCellsDoubleClick",
    "mergedCellsContextMenu",
    "mergedCellsMouseDown",
    "mergedCellsMouseUp",
    "mergedCellsMouseMove",
    // ============== Sort ====================
    "rangeSort",
    "rangeSorted",
    // ============== Filter ====================
    "rangeFilter",
    "rangeFiltered",
    // ============== Layout ====================
    "layoutAfterHeaderLayout",
    "layoutPagination",
    "beforeRender",
    "afterRender",
    "mounted",
    "destroy",
    // ============== Resize ====================
    "layoutResize",
    "layoutResizeSeriesWidth",
    "layoutResizeRowWidth",
    "layoutResizeRowHeight",
    "layoutResizeColWidth",
    "layoutResizeColHeight",
    "layoutResizeTreeWidth",
    "layoutResizeMouseDown",
    "layoutResizeMouseUp",
    "layoutResizeMouseMove",
    // ============== Global ====================
    "keyBoardDown",
    "keyBoardUp",
    "copied",
    "actionIconHover",
    "actionIconClick",
    "contextMenu",
    "mouseHover",
    "mouseUp",
    "mouseMove",
    "mouseDown",
    "selected",
    "reset",
    "linkFieldJump",
    "click",
    "doubleClick",
    "scroll",
    "hover",
    // ============== Auto 自动生成的 ================
    "layoutAfterRealDataCellRender",
    "rowCellBrushSelection",
    "colCellBrushSelection"
  ];
  return keys;
};
export {
  initBaseSheetEmits,
  initBaseSheetProps,
  initDrillDownEmits,
  initDrillDownProps
};
