// @antv/s2-vue/esm/index.js
import "./utils/extendLocale.js";
import { CustomTooltip } from "./components/tooltip/custom-tooltip.js";
import { default as default2 } from "./components/tooltip/index.vue.js";
import { default as default3 } from "./components/sheets/base-sheet.vue.js";
import { default as default4 } from "./components/sheets/index.vue.js";
import { default as default5 } from "./components/sheets/pivot-sheet.vue.js";
import { default as default6 } from "./components/sheets/table-sheet.vue.js";
import { default as default7 } from "./components/sheets/editable-sheet.vue.js";
import { default as default8 } from "./components/sheets/chart-sheet.vue.js";
import { default as default9 } from "./components/sheets/grid-analysis-sheet/index.vue.js";
import { default as default10 } from "./components/sheets/strategy-sheet/index.vue.js";
import { default as default11 } from "./components/sheets/pivot-chart-sheet.vue.js";
import { StrategySheetDataSet } from "./components/sheets/strategy-sheet/custom-data-set.js";
import { StrategySheetColCell } from "./components/sheets/strategy-sheet/custom-col-cell.js";
import { StrategySheetDataCell } from "./components/sheets/strategy-sheet/custom-data-cell.js";
import { default as default12 } from "./components/sheets/strategy-sheet/custom-tooltip/col-cell-tooltip.vue.js";
import { default as default13 } from "./components/sheets/strategy-sheet/custom-tooltip/data-cell-tooltip.vue.js";
import { default as default14 } from "./components/sheets/strategy-sheet/custom-tooltip/row-cell-tooltip.vue.js";
import { GridAnalysisSheetDataCell } from "./components/sheets/grid-analysis-sheet/custom-cell.js";
import { GridAnalysisTheme } from "./components/sheets/grid-analysis-sheet/theme.js";
import { useSpreadSheetInstance } from "./context/SpreadSheetContext.js";
import { useExpose } from "./hooks/useExpose.js";
const version = "@antv/s2-vue-v2.1.0";
export {
  default3 as BaseSheet,
  default8 as ChartSheet,
  CustomTooltip,
  default7 as EditableSheet,
  default9 as GridAnalysisSheet,
  GridAnalysisSheetDataCell,
  GridAnalysisTheme,
  default11 as PivotChartSheet,
  default5 as PivotSheet,
  default4 as SheetComponent,
  default10 as StrategySheet,
  StrategySheetColCell,
  default12 as StrategySheetColCellTooltip,
  StrategySheetDataCell,
  default13 as StrategySheetDataCellTooltip,
  StrategySheetDataSet,
  default14 as StrategySheetRowCellTooltip,
  default6 as TableSheet,
  default2 as TooltipComponent,
  useExpose,
  useSpreadSheetInstance,
  version
};
