// @antv/s2-vue/lib/components/tooltip/components/operator/index.vue2.js
"use strict";
const s2 = require("@antv/s2");
const antDesignVue = require("ant-design-vue");
const vue = require("vue");
const title = require("./title.vue2.js");
const menu = require("./menu.vue2.js");
const _sfc_main = vue.defineComponent({
  name: "TooltipOperator",
  props: [
    "menu",
    "onlyShowOperator",
    "cell"
  ],
  setup(props) {
    const { menu: menu2, cell } = props;
    const onMenuClick = (menuInfo) => {
      var _a;
      (_a = menu2 == null ? void 0 : menu2.onClick) == null ? void 0 : _a.call(menu2, menuInfo, cell);
    };
    const onMenuTitleClick = (subMenu) => () => {
      subMenu == null ? void 0 : subMenu.onClick(subMenu, cell);
    };
    return {
      TOOLTIP_PREFIX_CLS: s2.TOOLTIP_PREFIX_CLS,
      onMenuClick,
      onMenuTitleClick,
      menus: menu2 == null ? void 0 : menu2.items
    };
  },
  components: {
    Menu: antDesignVue.Menu,
    Dropdown: antDesignVue.Dropdown,
    TooltipOperatorTitle: title,
    TooltipOperatorMenu: menu
  }
});
module.exports = _sfc_main;
