// @antv/s2-vue/lib/components/tooltip/components/operator/menu.vue2.js
"use strict";
const menu_vue_vue_type_script_lang = require("./menu.vue.js");
const vue = require("vue");
const _pluginVue_exportHelper = require("../../../../_virtual/_plugin-vue_export-helper.js");
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_TooltipOperatorTitle = vue.resolveComponent("TooltipOperatorTitle");
  const _component_TooltipOperatorMenu = vue.resolveComponent("TooltipOperatorMenu");
  const _component_MenuItem = vue.resolveComponent("MenuItem");
  const _component_SubMenu = vue.resolveComponent("SubMenu");
  return vue.openBlock(), vue.createElementBlock(vue.Fragment, null, [
    !_ctx.isEmpty(_ctx.menu.children) ? (vue.openBlock(), vue.createBlock(_component_SubMenu, {
      key: _ctx.menu.key,
      popupClassName: `${_ctx.TOOLTIP_PREFIX_CLS}-operator-submenu-popup`,
      onTitleClick: _ctx.onMenuTitleClick
    }, {
      title: vue.withCtx(() => [
        vue.createVNode(_component_TooltipOperatorTitle, {
          menu: _ctx.menu,
          onClick: _ctx.onMenuTitleClick
        }, null, 8, ["menu", "onClick"])
      ]),
      default: vue.withCtx(() => [
        (vue.openBlock(true), vue.createElementBlock(vue.Fragment, null, vue.renderList(_ctx.menu.children, (subMenu) => {
          var _a;
          return vue.openBlock(), vue.createElementBlock(vue.Fragment, {
            key: subMenu.key
          }, [
            ((_a = subMenu == null ? void 0 : subMenu.children) == null ? void 0 : _a.length) ? (vue.openBlock(), vue.createBlock(_component_TooltipOperatorMenu, {
              key: 0,
              menu: subMenu,
              cell: _ctx.cell
            }, null, 8, ["menu", "cell"])) : (vue.openBlock(), vue.createBlock(_component_MenuItem, {
              title: subMenu.label,
              key: subMenu.key
            }, {
              default: vue.withCtx(() => [
                vue.createVNode(_component_TooltipOperatorTitle, {
                  menu: subMenu,
                  onClick: _ctx.onMenuTitleClick
                }, null, 8, ["menu", "onClick"])
              ]),
              _: 2
            }, 1032, ["title"]))
          ], 64);
        }), 128))
      ]),
      _: 1
    }, 8, ["popupClassName", "onTitleClick"])) : vue.createCommentVNode("", true),
    _ctx.isEmpty(_ctx.menu.children) ? (vue.openBlock(), vue.createBlock(_component_MenuItem, {
      title: _ctx.menu.label,
      key: _ctx.menu.key
    }, {
      default: vue.withCtx(() => [
        vue.createVNode(_component_TooltipOperatorTitle, {
          menu: _ctx.menu,
          onClick: _ctx.onMenuTitleClick
        }, null, 8, ["menu", "onClick"])
      ]),
      _: 1
    }, 8, ["title"])) : vue.createCommentVNode("", true)
  ], 64);
}
const TooltipOperatorMenu = /* @__PURE__ */ _pluginVue_exportHelper(menu_vue_vue_type_script_lang, [["render", _sfc_render]]);
module.exports = TooltipOperatorMenu;
