// @antv/s2-vue/esm/components/drill-down/index.vue2.js
import { defineComponent, ref } from "vue";
import { DRILL_DOWN_PRE_CLASS } from "@antv/s2";
import { MenuItem, Menu, Empty, Input, Button } from "ant-design-vue";
import { isEmpty } from "lodash";
import LocationIcon from "../../icons/location-icon.vue.js";
import TextIcon from "../../icons/text-icon.vue.js";
import CalendarIcon from "../../icons/calendar-icon.vue.js";
import { initDrillDownEmits, initDrillDownProps } from "../../utils/initPropAndEmits.js";
const _sfc_main = defineComponent({
  name: "DrillDown",
  props: initDrillDownProps(),
  emits: initDrillDownEmits(),
  methods: {},
  components: {
    Button,
    Input,
    Empty,
    Menu,
    MenuItem,
    LocationIcon,
    TextIcon,
    CalendarIcon
  },
  setup(props) {
    const {
      dataSet,
      disabledFields,
      getDrillFields,
      setDrillFields,
      className
    } = props;
    const getOptions = () => dataSet.map((val) => {
      val.disabled = !!(disabledFields && disabledFields.includes(val.value));
      return val;
    });
    const options = ref(getOptions());
    const selected = ref([]);
    const handleSearch = (e) => {
      const { value } = e.target;
      if (!value) {
        options.value = [...dataSet];
      } else {
        const reg = new RegExp(value, "gi");
        const result = dataSet.filter((item) => reg.test(item.name));
        options.value = [...result];
      }
    };
    const handleSelect = (value) => {
      const key = value == null ? void 0 : value.selectedKeys;
      selected.value = key;
      if (getDrillFields) {
        getDrillFields(key);
      }
      if (setDrillFields) {
        setDrillFields(key);
      }
    };
    const handleClear = (e) => {
      e.stopPropagation();
      selected.value = [];
      if (getDrillFields) {
        getDrillFields([]);
      }
      if (setDrillFields) {
        setDrillFields([]);
      }
    };
    return {
      options,
      handleSearch,
      handleSelect,
      handleClear,
      className,
      selected,
      isEmpty,
      DRILL_DOWN_PRE_CLASS
    };
  }
});
export {
  _sfc_main as default
};
