// @antv/s2-vue/esm/components/drill-down/index.vue.js
import _sfc_main from "./index.vue2.js";
import { resolveComponent, createElementBlock, openBlock, normalizeClass, createElementVNode, createVNode, createBlock, createCommentVNode, toDisplayString, withCtx, createTextVNode, Fragment, renderList } from "vue";
/* empty css           */
import _export_sfc from "../../_virtual/_plugin-vue_export-helper.js";
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_Button = resolveComponent("Button");
  const _component_Input = resolveComponent("Input");
  const _component_Empty = resolveComponent("Empty");
  const _component_TextIcon = resolveComponent("TextIcon");
  const _component_CalendarIcon = resolveComponent("CalendarIcon");
  const _component_LocationIcon = resolveComponent("LocationIcon");
  const _component_MenuItem = resolveComponent("MenuItem");
  const _component_Menu = resolveComponent("Menu");
  return openBlock(), createElementBlock("div", {
    class: normalizeClass([_ctx.DRILL_DOWN_PRE_CLASS, _ctx.className])
  }, [
    createElementVNode("header", {
      class: normalizeClass(`${_ctx.DRILL_DOWN_PRE_CLASS}-header`)
    }, [
      createElementVNode("div", null, toDisplayString(_ctx.title), 1),
      createVNode(_component_Button, {
        type: "link",
        onClick: _ctx.handleClear
      }, {
        default: withCtx(() => [
          createTextVNode(toDisplayString(_ctx.clearText), 1)
        ]),
        _: 1
      }, 8, ["onClick"])
    ], 2),
    createVNode(_component_Input, {
      class: normalizeClass(`${_ctx.DRILL_DOWN_PRE_CLASS}-search`),
      placeholder: _ctx.searchText,
      onChange: _ctx.handleSearch,
      onPressEnter: _ctx.handleSearch,
      allowClear: true
    }, null, 8, ["class", "placeholder", "onChange", "onPressEnter"]),
    _ctx.isEmpty(_ctx.options) ? (openBlock(), createBlock(_component_Empty, {
      key: 0,
      imageStyle: { height: "64px" },
      class: normalizeClass(`${_ctx.DRILL_DOWN_PRE_CLASS}-empty`)
    }, null, 8, ["class"])) : createCommentVNode("", true),
    createVNode(_component_Menu, {
      class: "`${DRILL_DOWN_PRE_CLASS}-menu`",
      selectedKeys: _ctx.selected,
      "onUpdate:selectedKeys": _cache[0] || (_cache[0] = ($event) => _ctx.selected = $event),
      onSelect: _ctx.handleSelect
    }, {
      default: withCtx(() => [
        (openBlock(true), createElementBlock(Fragment, null, renderList(_ctx.options, (option) => {
          return openBlock(), createBlock(_component_MenuItem, {
            key: option.value,
            disabled: option.disabled,
            class: normalizeClass(`${_ctx.DRILL_DOWN_PRE_CLASS}-menu-item`)
          }, {
            icon: withCtx(() => [
              option.type === "text" ? (openBlock(), createBlock(_component_TextIcon, { key: 0 })) : createCommentVNode("", true),
              option.type === "date" ? (openBlock(), createBlock(_component_CalendarIcon, { key: 1 })) : createCommentVNode("", true),
              option.type === "location" ? (openBlock(), createBlock(_component_LocationIcon, { key: 2 })) : createCommentVNode("", true)
            ]),
            default: withCtx(() => [
              createTextVNode(" " + toDisplayString(option == null ? void 0 : option.name), 1)
            ]),
            _: 2
          }, 1032, ["disabled", "class"]);
        }), 128))
      ]),
      _: 1
    }, 8, ["selectedKeys", "onSelect"])
  ], 2);
}
const DrillDown = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], ["__scopeId", "data-v-51804a8c"]]);
export {
  DrillDown as default
};
