// @antv/s2-vue/lib/hooks/useSpreadSheet.js
"use strict";
var __async = (__this, __arguments, generator) => {
  return new Promise((resolve, reject) => {
    var fulfilled = (value) => {
      try {
        step(generator.next(value));
      } catch (e) {
        reject(e);
      }
    };
    var rejected = (value) => {
      try {
        step(generator.throw(value));
      } catch (e) {
        reject(e);
      }
    };
    var step = (x) => x.done ? resolve(x.value) : Promise.resolve(x.value).then(fulfilled, rejected);
    step((generator = generator.apply(__this, __arguments)).next());
  });
};
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const s2 = require("@antv/s2");
const _extends = require("@antv/s2/extends");
const vue = require("vue");
const options = require("../utils/options.js");
const useEvents = require("./useEvents.js");
const useLoading = require("./useLoading.js");
const usePagination = require("./usePagination.js");
const useResize = require("./useResize.js");
const useSheetUpdate = require("./useSheetUpdate.js");
function useSpreadSheet(props, emit) {
  const {
    dataCfg,
    options: options$1,
    themeCfg,
    sheetType,
    spreadsheet: customSpreadSheet,
    onMounted: onS2Mounted
  } = props;
  const loadingRef = vue.toRef(props, "loading");
  const wrapperRef = vue.ref();
  const containerRef = vue.ref();
  const s2Ref = vue.shallowRef();
  const { loading, setLoading } = useLoading.useLoading(s2Ref, loadingRef);
  const pagination = usePagination.usePagination(s2Ref, props);
  const renderSpreadSheet = (container) => {
    const rawDataCfg = vue.toRaw(dataCfg);
    const rawOptions = vue.toRaw(options$1);
    const s2Options = options.getSheetComponentOptions(rawOptions);
    if (customSpreadSheet) {
      return customSpreadSheet(container, rawDataCfg, s2Options);
    }
    if (sheetType === "table" || sheetType === "editable") {
      return new s2.TableSheet(container, rawDataCfg, s2Options);
    }
    if (sheetType === "pivotChart") {
      return new _extends.PivotChartSheet(container, rawDataCfg, s2Options);
    }
    return new s2.PivotSheet(container, rawDataCfg, s2Options);
  };
  const buildSpreadSheet = () => __async(this, null, function* () {
    setLoading(true);
    s2Ref.value = renderSpreadSheet(containerRef.value);
    s2Ref.value.setThemeCfg(vue.toRaw(themeCfg));
    yield s2Ref.value.render();
    setLoading(false);
    onS2Mounted == null ? void 0 : onS2Mounted(s2Ref.value);
  });
  vue.onMounted(buildSpreadSheet);
  useEvents.useEvents(s2Ref, emit);
  useSheetUpdate.useSheetUpdate(s2Ref, props, {
    before() {
      setLoading(true);
    },
    after() {
      setLoading(false);
    }
  });
  useResize.useResize(s2Ref, props, { wrapperRef, containerRef });
  vue.onBeforeUnmount(() => {
    var _a;
    setLoading(false);
    (_a = s2Ref.value) == null ? void 0 : _a.destroy();
  });
  return {
    wrapperRef,
    containerRef,
    s2Ref,
    loading,
    setLoading,
    pagination
  };
}
exports.useSpreadSheet = useSpreadSheet;
