// @antv/s2-vue/esm/hooks/usePagination.js
var __async = (__this, __arguments, generator) => {
  return new Promise((resolve, reject) => {
    var fulfilled = (value) => {
      try {
        step(generator.next(value));
      } catch (e) {
        reject(e);
      }
    };
    var rejected = (value) => {
      try {
        step(generator.throw(value));
      } catch (e) {
        reject(e);
      }
    };
    var step = (x) => x.done ? resolve(x.value) : Promise.resolve(x.value).then(fulfilled, rejected);
    step((generator = generator.apply(__this, __arguments)).next());
  });
};
import { S2Event } from "@antv/s2";
import { isEmpty, omit } from "lodash";
import { ref, computed, watch } from "vue";
const DEFAULT_PAGE_NUMBER = 1;
const DEFAULT_PAGE_SIZE = 10;
const usePagination = (s2Ref, props) => {
  var _a, _b;
  const {
    options: { pagination: paginationCfg },
    showPagination
  } = props;
  const current = ref((_a = paginationCfg == null ? void 0 : paginationCfg.current) != null ? _a : DEFAULT_PAGE_NUMBER);
  const pageSize = ref((_b = paginationCfg == null ? void 0 : paginationCfg.pageSize) != null ? _b : DEFAULT_PAGE_SIZE);
  const total = ref(0);
  const change = (nextCurrent) => {
    current.value = nextCurrent;
  };
  const showSizeChange = (nextPageSize) => {
    pageSize.value = nextPageSize;
  };
  const visible = computed(() => {
    return showPagination && !isEmpty(paginationCfg);
  });
  watch([current, pageSize], () => __async(void 0, null, function* () {
    if (!s2Ref.value) {
      return;
    }
    const nextPagination = isEmpty(paginationCfg) ? null : {
      current: current.value,
      pageSize: pageSize.value
    };
    s2Ref.value.updatePagination(nextPagination);
    yield s2Ref.value.render(false);
  }));
  watch(
    [() => paginationCfg, s2Ref],
    () => {
      var _a2, _b2, _c, _d, _e;
      current.value = (_a2 = paginationCfg == null ? void 0 : paginationCfg.current) != null ? _a2 : DEFAULT_PAGE_NUMBER;
      pageSize.value = (_b2 = paginationCfg == null ? void 0 : paginationCfg.pageSize) != null ? _b2 : DEFAULT_PAGE_SIZE;
      total.value = (_e = (_d = (_c = s2Ref.value) == null ? void 0 : _c.facet) == null ? void 0 : _d.viewCellHeights.getTotalLength()) != null ? _e : 0;
    },
    {
      immediate: true
    }
  );
  watch(s2Ref, (value, oldValue, onCleanup) => {
    if (!s2Ref.value) {
      return;
    }
    const totalUpdateCallback = (data) => {
      if (isEmpty(paginationCfg)) {
        return;
      }
      total.value = data.total;
    };
    s2Ref.value.on(S2Event.LAYOUT_PAGINATION, totalUpdateCallback);
    onCleanup(() => {
      var _a2;
      (_a2 = s2Ref.value) == null ? void 0 : _a2.off(S2Event.LAYOUT_PAGINATION, totalUpdateCallback);
    });
  });
  return {
    customOptions: omit(paginationCfg, ["current", "pageSize", "total"]),
    visible,
    current,
    pageSize,
    total,
    change,
    showSizeChange
  };
};
export {
  DEFAULT_PAGE_NUMBER,
  DEFAULT_PAGE_SIZE,
  usePagination
};
