// @antv/s2-vue/esm/hooks/useSheetUpdate.js
var __async = (__this, __arguments, generator) => {
  return new Promise((resolve, reject) => {
    var fulfilled = (value) => {
      try {
        step(generator.next(value));
      } catch (e) {
        reject(e);
      }
    };
    var rejected = (value) => {
      try {
        step(generator.throw(value));
      } catch (e) {
        reject(e);
      }
    };
    var step = (x) => x.done ? resolve(x.value) : Promise.resolve(x.value).then(fulfilled, rejected);
    step((generator = generator.apply(__this, __arguments)).next());
  });
};
import { reactive, watch, isProxy } from "vue";
const useSheetUpdate = (s2Ref, props, hooks) => {
  const updateFlag = reactive({
    rerender: false,
    reloadData: false,
    rebuildDataset: false
  });
  watch(
    () => props.options,
    (options, prevOptions) => {
      var _a, _b;
      updateFlag.rerender = true;
      if (!Object.is(prevOptions == null ? void 0 : prevOptions.hierarchyType, options == null ? void 0 : options.hierarchyType)) {
        updateFlag.reloadData = true;
        updateFlag.rebuildDataset = true;
      }
      (_a = s2Ref.value) == null ? void 0 : _a.setOptions(options);
      (_b = s2Ref.value) == null ? void 0 : _b.changeSheetSize(options == null ? void 0 : options.width, options == null ? void 0 : options.height);
    },
    { deep: isProxy(props.options) }
  );
  watch(
    () => props.dataCfg,
    (dataCfg, prevDataCfg) => {
      var _a, _b, _c, _d, _e, _f, _g;
      if (((_b = (_a = prevDataCfg == null ? void 0 : prevDataCfg.fields) == null ? void 0 : _a.columns) == null ? void 0 : _b.length) !== ((_d = (_c = dataCfg == null ? void 0 : dataCfg.fields) == null ? void 0 : _c.columns) == null ? void 0 : _d.length)) {
        (_f = (_e = s2Ref.value) == null ? void 0 : _e.facet) == null ? void 0 : _f.clearInitColLeafNodes();
      }
      updateFlag.rerender = true;
      updateFlag.reloadData = true;
      (_g = s2Ref.value) == null ? void 0 : _g.setDataCfg(dataCfg);
    },
    { deep: isProxy(props.dataCfg) }
  );
  watch(
    () => props.themeCfg,
    (themeCfg) => {
      var _a;
      updateFlag.rerender = true;
      (_a = s2Ref.value) == null ? void 0 : _a.setThemeCfg(themeCfg);
    },
    {
      deep: isProxy(props.themeCfg)
    }
  );
  watch(updateFlag, (flag) => __async(void 0, null, function* () {
    var _a, _b, _c, _d, _e;
    if (!flag.rerender) {
      return;
    }
    (_a = hooks == null ? void 0 : hooks.before) == null ? void 0 : _a.call(hooks);
    const defaultRenderOptions = {
      reloadData: flag.reloadData,
      rebuildDataSet: flag.rebuildDataset
    };
    const renderOptions = ((_b = props == null ? void 0 : props.onUpdate) == null ? void 0 : _b.call(props, defaultRenderOptions)) || defaultRenderOptions;
    yield (_c = s2Ref.value) == null ? void 0 : _c.render(renderOptions);
    (_d = props == null ? void 0 : props.onUpdateAfterRender) == null ? void 0 : _d.call(props, renderOptions);
    (_e = hooks == null ? void 0 : hooks.after) == null ? void 0 : _e.call(hooks);
    flag.rerender = false;
    flag.reloadData = false;
    flag.rebuildDataset = false;
  }));
};
export {
  useSheetUpdate
};
