// @antv/s2-vue/esm/components/sheets/editable-sheet/drag-copy/DragCopyMask.vue2.js
import _sfc_main from "./DragCopyMask.vue.js";
import { createElementBlock, openBlock, normalizeStyle, normalizeClass } from "vue";
import _export_sfc from "../../../../_virtual/_plugin-vue_export-helper.js";
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("div", {
    class: normalizeClass(`${_ctx.S2_PREFIX_CLS}-drag-copy-mask`),
    style: normalizeStyle({
      right: _ctx.maskPosition.right > 0 ? `${_ctx.maskPosition.right}px` : "0",
      bottom: _ctx.maskPosition.bottom > 0 ? `${_ctx.maskPosition.bottom}px` : "0",
      left: _ctx.maskPosition.right <= 0 ? `${_ctx.maskPosition.right}px` : "0",
      top: _ctx.maskPosition.bottom <= 0 ? `${_ctx.maskPosition.bottom}px` : "0",
      width: `${Math.abs(_ctx.maskPosition.right)}px`,
      height: `${Math.abs(_ctx.maskPosition.bottom)}px`
    })
  }, null, 6);
}
const DragCopyMask = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  DragCopyMask as default
};
