// @antv/s2-vue/lib/components/tooltip/index.vue.js
"use strict";
const index_vue_vue_type_script_lang = require("./index.vue2.js");
const vue = require("vue");
;/* empty css            */
const _pluginVue_exportHelper = require("../../_virtual/_plugin-vue_export-helper.js");
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _a, _b, _c, _d, _e, _f, _g, _h, _i, _j, _k, _l, _m, _n, _o, _p, _q, _r, _s, _t, _u, _v, _w, _x;
  const _component_TooltipOperator = vue.resolveComponent("TooltipOperator");
  const _component_TooltipSimpleTips = vue.resolveComponent("TooltipSimpleTips");
  const _component_TooltipSummary = vue.resolveComponent("TooltipSummary");
  const _component_TooltipInterpretation = vue.resolveComponent("TooltipInterpretation");
  const _component_TooltipHeadInfo = vue.resolveComponent("TooltipHeadInfo");
  const _component_TooltipDetail = vue.resolveComponent("TooltipDetail");
  const _component_TooltipInfos = vue.resolveComponent("TooltipInfos");
  const _component_TooltipDescription = vue.resolveComponent("TooltipDescription");
  return _ctx.onlyShowOperator ? (vue.openBlock(), vue.createBlock(_component_TooltipOperator, {
    key: 0,
    menu: (_a = _ctx.operator) == null ? void 0 : _a.menu,
    onlyShowOperator: true,
    cell: _ctx.cell
  }, null, 8, ["menu", "cell"])) : (vue.openBlock(), vue.createElementBlock(vue.Fragment, { key: 1 }, [
    vue.createVNode(_component_TooltipOperator, {
      menu: (_b = _ctx.operator) == null ? void 0 : _b.menu,
      onlyShowOperator: false,
      cell: _ctx.cell
    }, null, 8, ["menu", "cell"]),
    _ctx.content ? vue.renderSlot(_ctx.$slots, "content", { key: 0 }) : (vue.openBlock(), vue.createElementBlock(vue.Fragment, { key: 1 }, [
      vue.createVNode(_component_TooltipSimpleTips, {
        name: (_c = _ctx.data) == null ? void 0 : _c.name,
        tips: (_d = _ctx.data) == null ? void 0 : _d.tips
      }, null, 8, ["name", "tips"]),
      ((_f = (_e = _ctx.data) == null ? void 0 : _e.summaries) == null ? void 0 : _f.length) ? (vue.openBlock(), vue.createBlock(_component_TooltipSummary, {
        key: 0,
        summaries: (_g = _ctx.data) == null ? void 0 : _g.summaries
      }, null, 8, ["summaries"])) : vue.createCommentVNode("", true),
      ((_h = _ctx.data) == null ? void 0 : _h.interpretation) ? (vue.openBlock(), vue.createBlock(_component_TooltipInterpretation, {
        key: 1,
        name: (_j = (_i = _ctx.data) == null ? void 0 : _i.interpretation) == null ? void 0 : _j.name,
        icon: (_l = (_k = _ctx.data) == null ? void 0 : _k.interpretation) == null ? void 0 : _l.icon,
        text: (_n = (_m = _ctx.data) == null ? void 0 : _m.interpretation) == null ? void 0 : _n.text,
        content: (_p = (_o = _ctx.data) == null ? void 0 : _o.interpretation) == null ? void 0 : _p.content
      }, null, 8, ["name", "icon", "text", "content"])) : vue.createCommentVNode("", true),
      vue.createVNode(_component_TooltipHeadInfo, {
        rows: ((_r = (_q = _ctx.data) == null ? void 0 : _q.headInfo) == null ? void 0 : _r.rows) || [],
        cols: ((_t = (_s = _ctx.data) == null ? void 0 : _s.headInfo) == null ? void 0 : _t.cols) || []
      }, null, 8, ["rows", "cols"]),
      vue.createVNode(_component_TooltipDetail, {
        list: ((_u = _ctx.data) == null ? void 0 : _u.details) || []
      }, null, 8, ["list"]),
      ((_v = _ctx.data) == null ? void 0 : _v.infos) ? (vue.openBlock(), vue.createBlock(_component_TooltipInfos, {
        key: 2,
        infos: (_w = _ctx.data) == null ? void 0 : _w.infos
      }, null, 8, ["infos"])) : vue.createCommentVNode("", true),
      vue.createVNode(_component_TooltipDescription, {
        description: (_x = _ctx.data) == null ? void 0 : _x.description
      }, null, 8, ["description"])
    ], 64))
  ], 64));
}
const TooltipComponent = /* @__PURE__ */ _pluginVue_exportHelper(index_vue_vue_type_script_lang, [["render", _sfc_render]]);
module.exports = TooltipComponent;
