// @antv/s2-vue/lib/components/tooltip/index.vue2.js
"use strict";
const s2 = require("@antv/s2");
const vue = require("vue");
const description = require("./components/description.vue.js");
const detail = require("./components/detail.vue.js");
const headInfo = require("./components/head-info.vue.js");
const infos = require("./components/infos.vue.js");
const interpretation = require("./components/interpretation.vue.js");
const simpleTips = require("./components/simple-tips.vue.js");
const summary = require("./components/summary.vue.js");
const index = require("./components/operator/index.vue.js");
require("./components/operator/menu.vue.js");
require("./components/operator/title.vue.js");
const _sfc_main = vue.defineComponent({
  name: "TooltipComponent",
  props: [
    "data",
    "options",
    "cell",
    "position",
    "event",
    "content"
  ],
  setup(props) {
    const { operator, onlyShowOperator } = s2.getTooltipDefaultOptions(
      props.options
    );
    return {
      operator,
      onlyShowOperator
    };
  },
  components: {
    TooltipDescription: description,
    TooltipDetail: detail,
    TooltipHeadInfo: headInfo,
    TooltipInfos: infos,
    TooltipInterpretation: interpretation,
    TooltipSimpleTips: simpleTips,
    TooltipSummary: summary,
    TooltipOperator: index
  }
});
module.exports = _sfc_main;
