// @antv/s2-vue/esm/components/sheets/grid-analysis-sheet/index.vue2.js
import { customMerge } from "@antv/s2";
import { defineComponent, toRefs, computed } from "vue";
import { useExpose } from "../../../hooks/useExpose.js";
import { initBaseSheetProps } from "../../../utils/initPropAndEmits.js";
import BaseSheet from "../base-sheet.vue.js";
import { GridAnalysisSheetDataCell } from "./custom-cell.js";
import { GridAnalysisTheme } from "./theme.js";
const _sfc_main = defineComponent({
  name: "GridAnalysisSheet",
  props: initBaseSheetProps(),
  emits: [],
  setup(props, ctx) {
    const s2Ref = useExpose(ctx.expose);
    const { options: defaultOptions, themeCfg } = toRefs(props);
    const s2Options = computed(() => {
      const options = {
        dataCell: (viewMeta, spreadsheet) => new GridAnalysisSheetDataCell(viewMeta, spreadsheet),
        showDefaultHeaderActionIcon: false,
        style: {
          colCell: {
            hideValue: true
          }
        }
      };
      return customMerge(defaultOptions == null ? void 0 : defaultOptions.value, options);
    });
    const s2ThemeCfg = computed(
      () => customMerge(themeCfg == null ? void 0 : themeCfg.value, { theme: GridAnalysisTheme })
    );
    return {
      s2Ref,
      s2Options,
      s2ThemeCfg
    };
  },
  components: {
    BaseSheet
  }
});
export {
  _sfc_main as default
};
