// @antv/s2-vue/esm/components/sheets/strategy-sheet/index.vue2.js
import { customMerge } from "@antv/s2";
import { isEmpty, size } from "lodash";
import { defineComponent, toRefs, computed } from "vue";
import { useExpose } from "../../../hooks/useExpose.js";
import { initBaseSheetProps } from "../../../utils/initPropAndEmits.js";
import BaseSheet from "../base-sheet.vue.js";
import { StrategySheetColCell } from "./custom-col-cell.js";
import { StrategySheetDataCell } from "./custom-data-cell.js";
import { StrategySheetDataSet } from "./custom-data-set.js";
const _sfc_main = defineComponent({
  name: "StrategySheet",
  props: initBaseSheetProps(),
  emits: [],
  setup(props, ctx) {
    const s2Ref = useExpose(ctx.expose);
    const { options, themeCfg, dataCfg } = toRefs(props);
    const strategySheetOptions = computed(() => {
      var _a, _b;
      if (isEmpty(dataCfg.value)) {
        return null;
      }
      const shouldHideValue = size((_b = (_a = dataCfg.value) == null ? void 0 : _a.fields) == null ? void 0 : _b.values) === 1;
      return {
        hierarchyType: "tree",
        dataCell: (viewMeta, spreadsheet) => new StrategySheetDataCell(viewMeta, spreadsheet),
        colCell: (node, spreadsheet, headerConfig) => new StrategySheetColCell(node, spreadsheet, headerConfig),
        dataSet: (spreadSheet) => new StrategySheetDataSet(spreadSheet),
        showDefaultHeaderActionIcon: false,
        style: {
          colCell: {
            hideValue: shouldHideValue
          }
        },
        interaction: {
          autoResetSheetStyle: true,
          // 趋势分析表禁用 刷选, 多选, 区间多选
          brushSelection: false,
          selectedCellMove: false,
          multiSelection: false,
          rangeSelection: false
        },
        tooltip: {
          operation: {
            hiddenColumns: true
          }
        }
      };
    });
    const s2Options = computed(
      () => customMerge(strategySheetOptions.value, options == null ? void 0 : options.value)
    );
    return {
      s2Ref,
      s2Options,
      themeCfg
    };
  },
  components: {
    BaseSheet
  }
});
export {
  _sfc_main as default
};
