// @antv/s2-vue/esm/components/sheets/strategy-sheet/custom-tooltip/col-cell-tooltip.vue2.js
import { getStrategySheetTooltipClsName } from "@antv/s2";
import { isFunction } from "lodash";
import { defineComponent, computed } from "vue";
const _sfc_main = defineComponent({
  name: "StrategySheetColCellTooltip",
  props: {
    cell: {
      type: Object,
      required: true
    },
    label: {
      type: [Object, Function],
      default: void 0
    }
  },
  setup(props) {
    const meta = computed(() => props.cell.getMeta());
    const shouldHide = computed(() => {
      return meta.value.isLeaf && meta.value.level !== 0;
    });
    const cellName = computed(() => {
      const m = meta.value;
      const name = m.spreadsheet.dataSet.getFieldName(m.field);
      const customLabel = isFunction(props.label) ? props.label(props.cell, name) : props.label;
      return customLabel != null ? customLabel : name;
    });
    const cellValue = computed(() => meta.value.value);
    return {
      tooltipCls: getStrategySheetTooltipClsName,
      shouldHide,
      cellName,
      cellValue
    };
  }
});
export {
  _sfc_main as default
};
