// @antv/s2-vue/esm/components/sheets/strategy-sheet/custom-tooltip/data-cell-tooltip.vue2.js
import { getEmptyPlaceholder, i18n, getStrategySheetTooltipClsName, isUnchangedValue, isUpDataValue } from "@antv/s2";
import { isFunction, first, get, isNil, isEmpty } from "lodash";
import { defineComponent, computed } from "vue";
const _sfc_main = defineComponent({
  name: "StrategySheetDataCellTooltip",
  props: {
    cell: {
      type: Object,
      required: true
    },
    label: {
      type: [Object, Function],
      default: void 0
    },
    showOriginalValue: {
      type: Boolean,
      default: false
    },
    renderDerivedValue: {
      type: Function,
      default: void 0
    }
  },
  setup(props) {
    const meta = computed(() => props.cell.getMeta());
    const spreadsheet = computed(() => meta.value.spreadsheet);
    const metaFieldValue = computed(
      () => {
        var _a;
        return (_a = meta.value) == null ? void 0 : _a.fieldValue;
      }
    );
    const rowDescription = computed(
      () => spreadsheet.value.dataSet.getCustomFieldDescription(props.cell)
    );
    const rowName = computed(() => {
      const defaultRowName = spreadsheet.value.dataSet.getCustomRowFieldName(
        props.cell
      );
      const customLabel = isFunction(props.label) ? props.label(props.cell, defaultRowName) : props.label;
      return customLabel != null ? customLabel : defaultRowName;
    });
    const colLeafNode = computed(
      () => spreadsheet.value.facet.getColLeafNodeByIndex(meta.value.colIndex)
    );
    const derivedLabels = computed(() => {
      var _a;
      try {
        const [, ...labels] = JSON.parse((_a = colLeafNode.value) == null ? void 0 : _a.value);
        return labels;
      } catch (e) {
        return [];
      }
    });
    const valuesCfg = computed(
      () => {
        var _a, _b;
        return (_b = (_a = spreadsheet.value.options.style) == null ? void 0 : _a.dataCell) == null ? void 0 : _b.valuesCfg;
      }
    );
    const values = computed(() => {
      var _a;
      const firstValues = first((_a = metaFieldValue.value) == null ? void 0 : _a.values) || [
        metaFieldValue.value
      ];
      return {
        value: firstValues[0],
        derivedValues: firstValues.slice(1)
      };
    });
    const originalValues = computed(() => {
      var _a;
      const originalValuesField = get(
        metaFieldValue.value,
        (_a = valuesCfg.value) == null ? void 0 : _a.originalValueField
      );
      const firstOriginal = first(originalValuesField) || [values.value.value];
      return {
        originalValue: firstOriginal[0],
        derivedOriginalValues: firstOriginal.slice(1)
      };
    });
    const emptyPlaceholder = computed(
      () => getEmptyPlaceholder(meta.value, spreadsheet.value.options.placeholder)
    );
    const shouldShowOriginalValue = computed(
      () => {
        var _a;
        return ((_a = valuesCfg.value) == null ? void 0 : _a.showOriginalValue) || props.showOriginalValue;
      }
    );
    const getDerivedValueInfo = (derivedValue, index) => {
      const isUnchanged = isUnchangedValue(
        derivedValue,
        values.value.value
      );
      const isUp = !isUnchanged && isUpDataValue(derivedValue);
      const isDown = !isUnchanged && !isUp;
      const originalDerivedValue = originalValues.value.derivedOriginalValues[index];
      return { isUnchanged, isUp, isDown, originalDerivedValue };
    };
    return {
      tooltipCls: getStrategySheetTooltipClsName,
      i18n,
      isEmpty,
      isNil,
      rowName,
      rowDescription,
      values,
      originalValues,
      derivedLabels,
      emptyPlaceholder,
      shouldShowOriginalValue,
      getDerivedValueInfo
    };
  }
});
export {
  _sfc_main as default
};
