// @antv/s2-vue/esm/components/sheets/strategy-sheet/custom-tooltip/row-cell-tooltip.vue2.js
import { i18n, getStrategySheetTooltipClsName } from "@antv/s2";
import { isFunction } from "lodash";
import { defineComponent, computed } from "vue";
const _sfc_main = defineComponent({
  name: "StrategySheetRowCellTooltip",
  props: {
    cell: {
      type: Object,
      required: true
    },
    label: {
      type: [Object, Function],
      default: void 0
    }
  },
  setup(props) {
    const meta = computed(() => props.cell.getMeta());
    const rowName = computed(() => {
      const { value } = meta.value;
      const customLabel = isFunction(props.label) ? props.label(props.cell, value) : props.label;
      return customLabel != null ? customLabel : value;
    });
    const description = computed(() => {
      const { field, spreadsheet, extra } = meta.value;
      return spreadsheet.dataSet.getFieldDescription(field) || (extra == null ? void 0 : extra["description"]);
    });
    return {
      tooltipCls: getStrategySheetTooltipClsName,
      i18n,
      rowName,
      description
    };
  }
});
export {
  _sfc_main as default
};
