// @antv/s2-vue/lib/components/sheets/base-sheet.vue.js
"use strict";
const baseSheet_vue_vue_type_script_lang = require("./base-sheet.vue2.js");
const vue = require("vue");
;/* empty css                 */
const _pluginVue_exportHelper = require("../../_virtual/_plugin-vue_export-helper.js");
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_S2Pagination = vue.resolveComponent("S2Pagination");
  const _component_Spin = vue.resolveComponent("Spin");
  return vue.openBlock(), vue.createBlock(_component_Spin, {
    wrapperClassName: _ctx.S2_PREFIX_CLS + "-spin",
    spinning: _ctx.loading
  }, {
    default: vue.withCtx(() => [
      vue.createElementVNode("div", {
        ref: "wrapperRef",
        class: vue.normalizeClass(_ctx.S2_PREFIX_CLS + "-wrapper")
      }, [
        vue.createElementVNode("div", {
          ref: "containerRef",
          class: vue.normalizeClass(_ctx.S2_PREFIX_CLS + "-container")
        }, null, 2),
        _ctx.pagination.visible.value ? (vue.openBlock(), vue.createBlock(_component_S2Pagination, {
          key: 0,
          current: _ctx.pagination.current.value,
          pageSize: _ctx.pagination.pageSize.value,
          total: _ctx.pagination.total.value,
          customOptions: _ctx.pagination.customOptions,
          onChange: _ctx.handlePageChange,
          onShowSizeChange: _ctx.handlePageSizeChange
        }, null, 8, ["current", "pageSize", "total", "customOptions", "onChange", "onShowSizeChange"])) : vue.createCommentVNode("", true),
        vue.renderSlot(_ctx.$slots, "editCell")
      ], 2)
    ]),
    _: 3
  }, 8, ["wrapperClassName", "spinning"]);
}
const BaseSheet = /* @__PURE__ */ _pluginVue_exportHelper(baseSheet_vue_vue_type_script_lang, [["render", _sfc_render]]);
module.exports = BaseSheet;
