// @antv/s2-vue/lib/components/sheets/base-sheet.vue2.js
"use strict";
const s2 = require("@antv/s2");
const antDesignVue = require("ant-design-vue");
const vue = require("vue");
const lodash = require("lodash");
const index = require("../pagination/index.vue.js");
const SpreadSheetContext = require("../../context/SpreadSheetContext.js");
const useSpreadSheet = require("../../hooks/useSpreadSheet.js");
const initPropAndEmits = require("../../utils/initPropAndEmits.js");
const _sfc_main = vue.defineComponent({
  name: "BaseSheet",
  props: initPropAndEmits.initBaseSheetProps(),
  emits: initPropAndEmits.initBaseSheetEmits(),
  setup(props, ctx) {
    const { wrapperRef, containerRef, s2Ref, loading, pagination } = useSpreadSheet.useSpreadSheet(props, ctx.emit);
    const s2RefComputed = vue.computed(() => {
      var _a;
      return (_a = s2Ref.value) != null ? _a : null;
    });
    SpreadSheetContext.provideSpreadSheet(s2RefComputed);
    ctx.expose({
      get instance() {
        return s2Ref.value;
      }
    });
    const handlePageChange = (nextCurrent) => {
      var _a, _b;
      if (props.showPagination && !lodash.isBoolean(props.showPagination)) {
        (_b = (_a = props.showPagination).onChange) == null ? void 0 : _b.call(_a, nextCurrent);
      }
      pagination.change(nextCurrent);
    };
    const handlePageSizeChange = (nextSize) => {
      var _a, _b;
      if (props.showPagination && !lodash.isBoolean(props.showPagination)) {
        (_b = (_a = props.showPagination).onShowSizeChange) == null ? void 0 : _b.call(_a, nextSize);
      }
      pagination.showSizeChange(nextSize);
    };
    return {
      S2_PREFIX_CLS: s2.S2_PREFIX_CLS,
      wrapperRef,
      containerRef,
      s2Ref,
      loading,
      pagination,
      handlePageChange,
      handlePageSizeChange
    };
  },
  components: {
    Spin: antDesignVue.Spin,
    S2Pagination: index
  }
});
module.exports = _sfc_main;
