// @antv/s2-vue/lib/components/sheets/chart-sheet.vue2.js
"use strict";
const s2 = require("@antv/s2");
const vue = require("vue");
const useExpose = require("../../hooks/useExpose.js");
const initPropAndEmits = require("../../utils/initPropAndEmits.js");
const baseSheet = require("./base-sheet.vue.js");
const _sfc_main = vue.defineComponent({
  name: "ChartSheet",
  props: initPropAndEmits.initBaseSheetProps(),
  emits: [],
  setup(props, ctx) {
    const s2Ref = useExpose.useExpose(ctx.expose);
    const { options: chartOptions, themeCfg: customThemeCfg } = vue.toRefs(props);
    const s2Options = vue.computed(() => {
      const defaultOptions = {
        style: {
          rowCell: {
            width: 100
          },
          dataCell: {
            width: 400,
            height: 400
          }
        },
        tooltip: {
          enable: true
        }
      };
      const options = {
        showDefaultHeaderActionIcon: false,
        interaction: {
          hoverFocus: false,
          brushSelection: {
            dataCell: false
          }
        }
      };
      return s2.customMerge(
        defaultOptions,
        chartOptions.value,
        options
      );
    });
    const themeCfg = vue.computed(() => {
      const defaultTheme = {
        dataCell: {
          cell: {
            interactionState: {
              hoverFocus: {
                borderOpacity: 0
              },
              selected: {
                borderOpacity: 0
              }
            }
          }
        }
      };
      return s2.customMerge(defaultTheme, customThemeCfg == null ? void 0 : customThemeCfg.value);
    });
    return {
      s2Ref,
      s2Options,
      themeCfg
    };
  },
  components: {
    BaseSheet: baseSheet
  }
});
module.exports = _sfc_main;
