// @antv/s2-vue/lib/components/sheets/editable-sheet.vue.js
"use strict";
const editableSheet_vue_vue_type_script_lang = require("./editable-sheet.vue2.js");
const vue = require("vue");
const _pluginVue_exportHelper = require("../../_virtual/_plugin-vue_export-helper.js");
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_Input = vue.resolveComponent("Input");
  const _component_DragCopyPoint = vue.resolveComponent("DragCopyPoint");
  const _component_BaseSheet = vue.resolveComponent("BaseSheet");
  return vue.openBlock(), vue.createBlock(_component_BaseSheet, vue.mergeProps({
    onDataCellDoubleClick: _ctx.onDataCellDbClick,
    onScroll: _ctx.setInputStyle,
    onDataCellClick: _ctx.onSave,
    ref: "s2Ref"
  }, _ctx.$props, { options: _ctx.options }), {
    editCell: vue.withCtx(() => [
      vue.createVNode(_component_Input, {
        onBlur: _ctx.onSave,
        value: _ctx.inputValue,
        ref: "inputRef",
        style: vue.normalizeStyle(_ctx.inputStyle)
      }, null, 8, ["onBlur", "value", "style"]),
      vue.createVNode(_component_DragCopyPoint)
    ]),
    _: 1
  }, 16, ["onDataCellDoubleClick", "onScroll", "onDataCellClick", "options"]);
}
const EditableSheet = /* @__PURE__ */ _pluginVue_exportHelper(editableSheet_vue_vue_type_script_lang, [["render", _sfc_render]]);
module.exports = EditableSheet;
