// @antv/s2-vue/lib/components/sheets/index.vue2.js
"use strict";
const vue = require("vue");
const useExpose = require("../../hooks/useExpose.js");
const chartSheet = require("./chart-sheet.vue.js");
const index$1 = require("./grid-analysis-sheet/index.vue.js");
const pivotChartSheet = require("./pivot-chart-sheet.vue.js");
const pivotSheet = require("./pivot-sheet.vue.js");
const index = require("./strategy-sheet/index.vue.js");
const tableSheet = require("./table-sheet.vue.js");
const editableSheet = require("./editable-sheet.vue.js");
const _sfc_main = vue.defineComponent({
  name: "Sheet",
  props: [],
  emits: [],
  setup(props, ctx) {
    const s2Ref = useExpose.useExpose(ctx.expose);
    const sheetType = vue.computed(() => {
      const type = ctx.attrs["sheetType"];
      switch (type) {
        case "table":
          return tableSheet;
        case "editable":
          return editableSheet;
        case "chart":
          return chartSheet;
        case "gridAnalysis":
          return index$1;
        case "strategy":
          return index;
        case "pivotChart":
          return pivotChartSheet;
        default:
          return pivotSheet;
      }
    });
    return { s2Ref, sheetType };
  },
  components: {
    PivotSheet: pivotSheet,
    TableSheet: tableSheet,
    EditableSheet: editableSheet,
    ChartSheet: chartSheet,
    GridAnalysisSheet: index$1,
    StrategySheet: index,
    PivotChartSheet: pivotChartSheet
  }
});
module.exports = _sfc_main;
