// @antv/s2-vue/lib/components/sheets/strategy-sheet/custom-data-set.js
"use strict";
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const s2 = require("@antv/s2");
const lodash = require("lodash");
class StrategySheetDataSet extends s2.CustomGridPivotDataSet {
  getExistValuesByDataItem(data) {
    const result = lodash.keys(data).filter((key) => lodash.isObject(data[key]));
    if (lodash.isEmpty(result)) {
      result.push(s2.EMPTY_EXTRA_FIELD_PLACEHOLDER);
    }
    return result;
  }
  processDataCfg(dataCfg) {
    const updatedDataCfg = super.processDataCfg(dataCfg);
    const newMeta = this.processMeta(dataCfg.meta, s2.i18n("数值"));
    return __spreadProps(__spreadValues({}, updatedDataCfg), {
      meta: newMeta,
      fields: __spreadProps(__spreadValues({}, updatedDataCfg.fields), {
        rows: [...dataCfg.fields.rows || [], s2.EXTRA_FIELD]
      })
    });
  }
}
exports.StrategySheetDataSet = StrategySheetDataSet;
