// @antv/s2-vue/lib/components/sheets/strategy-sheet/index.vue2.js
"use strict";
const s2 = require("@antv/s2");
const lodash = require("lodash");
const vue = require("vue");
const useExpose = require("../../../hooks/useExpose.js");
const initPropAndEmits = require("../../../utils/initPropAndEmits.js");
const baseSheet = require("../base-sheet.vue.js");
const customColCell = require("./custom-col-cell.js");
const customDataCell = require("./custom-data-cell.js");
const customDataSet = require("./custom-data-set.js");
const _sfc_main = vue.defineComponent({
  name: "StrategySheet",
  props: initPropAndEmits.initBaseSheetProps(),
  emits: [],
  setup(props, ctx) {
    const s2Ref = useExpose.useExpose(ctx.expose);
    const { options, themeCfg, dataCfg } = vue.toRefs(props);
    const strategySheetOptions = vue.computed(() => {
      var _a, _b;
      if (lodash.isEmpty(dataCfg.value)) {
        return null;
      }
      const shouldHideValue = lodash.size((_b = (_a = dataCfg.value) == null ? void 0 : _a.fields) == null ? void 0 : _b.values) === 1;
      return {
        hierarchyType: "tree",
        dataCell: (viewMeta, spreadsheet) => new customDataCell.StrategySheetDataCell(viewMeta, spreadsheet),
        colCell: (node, spreadsheet, headerConfig) => new customColCell.StrategySheetColCell(node, spreadsheet, headerConfig),
        dataSet: (spreadSheet) => new customDataSet.StrategySheetDataSet(spreadSheet),
        showDefaultHeaderActionIcon: false,
        style: {
          colCell: {
            hideValue: shouldHideValue
          }
        },
        interaction: {
          autoResetSheetStyle: true,
          // 趋势分析表禁用 刷选, 多选, 区间多选
          brushSelection: false,
          selectedCellMove: false,
          multiSelection: false,
          rangeSelection: false
        },
        tooltip: {
          operation: {
            hiddenColumns: true
          }
        }
      };
    });
    const s2Options = vue.computed(
      () => s2.customMerge(strategySheetOptions.value, options == null ? void 0 : options.value)
    );
    return {
      s2Ref,
      s2Options,
      themeCfg
    };
  },
  components: {
    BaseSheet: baseSheet
  }
});
module.exports = _sfc_main;
