// @antv/s2-vue/lib/components/drill-down/index.vue2.js
"use strict";
const vue = require("vue");
const s2 = require("@antv/s2");
const antDesignVue = require("ant-design-vue");
const lodash = require("lodash");
const locationIcon = require("../../icons/location-icon.vue.js");
const textIcon = require("../../icons/text-icon.vue.js");
const calendarIcon = require("../../icons/calendar-icon.vue.js");
const initPropAndEmits = require("../../utils/initPropAndEmits.js");
const _sfc_main = vue.defineComponent({
  name: "DrillDown",
  props: initPropAndEmits.initDrillDownProps(),
  emits: initPropAndEmits.initDrillDownEmits(),
  methods: {},
  components: {
    Button: antDesignVue.Button,
    Input: antDesignVue.Input,
    Empty: antDesignVue.Empty,
    Menu: antDesignVue.Menu,
    MenuItem: antDesignVue.MenuItem,
    LocationIcon: locationIcon,
    TextIcon: textIcon,
    CalendarIcon: calendarIcon
  },
  setup(props) {
    const {
      dataSet,
      disabledFields,
      getDrillFields,
      setDrillFields,
      className
    } = props;
    const getOptions = () => dataSet.map((val) => {
      val.disabled = !!(disabledFields && disabledFields.includes(val.value));
      return val;
    });
    const options = vue.ref(getOptions());
    const selected = vue.ref([]);
    const handleSearch = (e) => {
      const { value } = e.target;
      if (!value) {
        options.value = [...dataSet];
      } else {
        const reg = new RegExp(value, "gi");
        const result = dataSet.filter((item) => reg.test(item.name));
        options.value = [...result];
      }
    };
    const handleSelect = (value) => {
      const key = value == null ? void 0 : value.selectedKeys;
      selected.value = key;
      if (getDrillFields) {
        getDrillFields(key);
      }
      if (setDrillFields) {
        setDrillFields(key);
      }
    };
    const handleClear = (e) => {
      e.stopPropagation();
      selected.value = [];
      if (getDrillFields) {
        getDrillFields([]);
      }
      if (setDrillFields) {
        setDrillFields([]);
      }
    };
    return {
      options,
      handleSearch,
      handleSelect,
      handleClear,
      className,
      selected,
      isEmpty: lodash.isEmpty,
      DRILL_DOWN_PRE_CLASS: s2.DRILL_DOWN_PRE_CLASS
    };
  }
});
module.exports = _sfc_main;
