// @antv/s2-vue/lib/components/pagination/index.vue.js
"use strict";
const index_vue_vue_type_script_lang = require("./index.vue2.js");
const vue = require("vue");
const _pluginVue_exportHelper = require("../../_virtual/_plugin-vue_export-helper.js");
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_AntDPagination = vue.resolveComponent("AntDPagination");
  return vue.openBlock(), vue.createElementBlock("div", {
    class: vue.normalizeClass(_ctx.PRE_CLASS)
  }, [
    vue.createVNode(_component_AntDPagination, vue.mergeProps({
      size: "small",
      "default-current": 1,
      showSizeChanger: true,
      showQuickJumper: _ctx.showQuickJumper
    }, _ctx.customOptions, {
      current: _ctx.current,
      total: _ctx.total,
      pageSize: _ctx.pageSize,
      onShowSizeChange: _cache[0] || (_cache[0] = (_, size) => _ctx.$emit("showSizeChange", size)),
      onChange: _cache[1] || (_cache[1] = (current) => _ctx.$emit("change", current))
    }), null, 16, ["showQuickJumper", "current", "total", "pageSize"]),
    vue.createElementVNode("span", {
      class: vue.normalizeClass(`${_ctx.PRE_CLASS}-count`)
    }, vue.toDisplayString(_ctx.i18n("共计")) + " " + vue.toDisplayString(_ctx.total || " - ") + " " + vue.toDisplayString(_ctx.i18n("条")), 3)
  ], 2);
}
const S2Pagination = /* @__PURE__ */ _pluginVue_exportHelper(index_vue_vue_type_script_lang, [["render", _sfc_render]]);
module.exports = S2Pagination;
